regex.go

```diff
@@ -48,13 +48,23 @@ type Regexp struct {
 }
 
 func NewRegexp(pattern string, option int) (re *Regexp, err error) {
-	re = &Regexp{pattern: pattern}
-	patternCharPtr := C.CString(pattern)
-	defer C.free(unsafe.Pointer(patternCharPtr))
+	re = &Regexp{pattern: pattern, encoding: C.ONIG_ENCODING_UTF8}
```

Comment:
It would be better not to use the named returns here:

```go
return initRegexp(&Regexp{pattern: pattern, encoding: C.ONIG_ENCODING_UTF8}, option)
```

The names are pre-existing but I also suggest dropping them, since they don't add any documentary value.

Reply:
thank you! addressed in e928d1f